### PR TITLE
[Feature:InstructorUI] Added course_section_id column

### PR DESCRIPTION
**Pull request by tkoz0**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
No `course_id` column in the database.

### What is the new behavior?
Added `course_id` column to the database to store an extra string for university-specific course (section) identifying information.

### Other information?

Step 1 of #8770. Tested by performing insert/update operations in the main Submitty database.

If you previously installed this migration, your databases will have extra `course_id` columns that will not be removed. It may be desirable to drop those columns now that this was renamed to `course_section_id`

### CODE REVIEW

**Review by kraduc — CHANGES_REQUESTED** [review]
Main review is the comment about dropping the column.

I'm not 100% sure if we need to worry about a possible state in which the sync migration is up and the adding column migration is down. If we do, these should probably be combined into one migration with two sql execute statements for the up migration and for the down migration. Otherwise, it is probably fine as it is.

File: migration/migrator/migrations/course/20230126130456_add_course_id_column.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+
+    database.execute("""
+ALTER TABLE IF EXISTS sections_registration
+DROP COLUMN IF EXISTS course_id
```
[the remark this answers is not in the document]
Comment (pbailie) [reply migration/migrator/migrations/course/20230126130456_add_course_id_column.py:42]: This information is available from the registrar, so its loss could be recovered.

File: migration/migrator/migrations/master/20230126125745_add_course_id_column.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    """
+
+    database.execute("""
+ALTER TABLE IF EXISTS courses_registration_sections
+DROP COLUMN IF EXISTS course_id
```
[the remark this answers is not in the document]
Comment (pbailie) [reply migration/migrator/migrations/master/20230126125745_add_course_id_column.py:34]: This information is available from the registrar, so data loss can be recovered.

**Comment by bmcutler** [comment]
@pbailie I agree, let's name it course_section_id

**Review by pbailie — APPROVED** [review]

**Pull request merged**